src/notification/NotificationController.ts

```diff
@@ -0,0 +1,266 @@
+import type { Patch } from 'immer';
+import { nanoid } from 'nanoid';
+
+import { hasProperty } from '../util';
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type NotificationState
+ * @property notifications - Stores existing notifications to be shown in the UI
+ */
+export type NotificationState = {
+  notifications: Record<string, Notification>;
+};
+
+export type Notification = {
```

Comment:
This type could use documentation.

The only reason our linter doesn't demand this is that the ESLint JSDoc plugin doesn't have very many TypeScript-specific rules right now, so this wasn't an option (as far as I could tell).